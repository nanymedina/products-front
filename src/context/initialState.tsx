import StateT from '@src/type/state';

const initialState: StateT = {
  products: [],
  search: null,
  shoppingCart: [],
  error: null,
};

export default initialState;
