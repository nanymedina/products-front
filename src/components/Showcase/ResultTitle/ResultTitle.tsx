import { Container } from 'react-bootstrap';
import { ReactElement } from 'react';

import { useStateApp } from '@src/context/appContext';
import Style from './ResultTitle.module.scss';

const ResultTitle = (): ReactElement => {
  const { products, search, error } = useStateApp();
  const foundResult = !!search && products.length >= 1;
  const notFoundResult = !!search && products.length === 0;
  const resultTitle = () => {
    return (
      <div data-testid="found-result">
        <span>Resultados para </span>
        <strong>{search}</strong>
        <span>:</span>
      </div>
    );
  };

  const titleWithoutResult = () => {
    return (
      <div data-testid="not-found-result">
        <span>No encontramos resultados para </span>
        <strong>{search}</strong>
        <span>:</span>
      </div>
    );
  };

  const titleError = () => {
    return (
      <div data-testid="title-error">
        <span>Ups algo salio mal!!!</span>
      </div>
    );
  };
  const welcomeTitle = () => {
    return (
      <div data-testid="welcome-title">
        <span>Bienvenido!!!</span>
      </div>
    );
  };

  let resultTitleMessage = welcomeTitle();

  if (notFoundResult) resultTitleMessage = titleWithoutResult();
  if (foundResult) resultTitleMessage = resultTitle();
  if (error) resultTitleMessage = titleError();

  return (
    <Container data-testid="result-title" className={Style.resul_title} fluid>
      {resultTitleMessage}
    </Container>
  );
};

export default ResultTitle;
