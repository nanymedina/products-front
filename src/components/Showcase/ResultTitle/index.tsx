import ResultTitle from './ResultTitle';

export default ResultTitle;
