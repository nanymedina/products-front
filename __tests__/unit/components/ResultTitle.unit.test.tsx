import { render, screen } from '@testing-library/react';
import { useStateApp } from '@src/context/appContext';
import ResultTitle from '@components/Showcase/ResultTitle';

jest.mock('@src/context/appContext');

describe('Result Title', () => {
  const useStateAppMock = useStateApp as jest.Mock;

  afterEach(() => {
    useStateAppMock.mockClear();
  });

  it('should displays welcome message on home page', () => {
    useStateAppMock.mockReturnValue({ products: [], search: null, error: null });

    render(<ResultTitle />);

    expect(screen.getByTestId('welcome-title').textContent).toEqual('Bienvenido!!!');
    expect(screen.getByTestId('result-title')).toMatchSnapshot();
  });

  it('should displays a message with the search result', () => {
    useStateAppMock.mockReturnValue({
      products: [{ product: 'product' }],
      search: 'search',
      error: null,
    });

    render(<ResultTitle />);

    expect(screen.getByTestId('found-result').textContent).toEqual('Resultados para search:');
    expect(screen.getByTestId('result-title')).toMatchSnapshot();
  });

  it('should displays a message with no search results', () => {
    useStateAppMock.mockReturnValue({
      products: [],
      search: 'search',
      error: null,
    });

    render(<ResultTitle />);

    expect(screen.getByTestId('not-found-result').textContent).toEqual(
      'No encontramos resultados para search:'
    );
    expect(screen.getByTestId('result-title')).toMatchSnapshot();
  });

  it('should displays an error message when a search error has occurred', () => {
    useStateAppMock.mockReturnValue({
      products: [],
      search: null,
      error: 'error',
    });

    render(<ResultTitle />);

    expect(screen.getByTestId('title-error').textContent).toEqual('Ups algo salio mal!!!');
    expect(screen.getByTestId('result-title')).toMatchSnapshot();
  });
});
