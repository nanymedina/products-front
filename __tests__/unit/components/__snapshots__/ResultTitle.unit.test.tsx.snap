// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Title show message error when has error 1`] = `
<div
  class="container-fluid"
  data-testid="result-title"
>
  <div
    data-testid="title-error"
  >
    <span>
      Ups algo salio mal!!!
    </span>
  </div>
</div>
`;

exports[`Result Title show message result when found result 1`] = `
<div
  class="container-fluid"
  data-testid="result-title"
>
  <div
    data-testid="found-result"
  >
    <span>
      Resultados para 
    </span>
    <strong>
      search
    </strong>
    <span>
      :
    </span>
  </div>
</div>
`;

exports[`Result Title show message the welcome when inital page 1`] = `
<div
  class="container-fluid"
  data-testid="result-title"
>
  <div
    data-testid="welcome-title"
  >
    <span>
      Bienvenido!!!
    </span>
  </div>
</div>
`;

exports[`Result Title show message without result when not found result 1`] = `
<div
  class="container-fluid"
  data-testid="result-title"
>
  <div
    data-testid="not-found-result"
  >
    <span>
      No encontramos resultados para 
    </span>
    <strong>
      search
    </strong>
    <span>
      :
    </span>
  </div>
</div>
`;
